### PR TITLE
Allow user to pass multipart param through to Fog

### PR DESCRIPTION
**Pull request by henders**
Add in multipart upload support for uploading files >5GB to AWS S3.

This should be mostly harmless change. I think the other fog stores pick out their own options. 

/cc @yadavsaroj @staugaard @pswadi-zendesk @stubotnik @grosser @steved

### CODE REVIEW

File: lib/remote_files/fog_store.rb [hunk 1]
```diff
@@ -98,5 +95,28 @@ def create_directory
...
+    def store_options(file)
+      store_options =
+        {
+          :body => file.content,
+          :content_type => file.content_type,
+          :key => file.identifier,
+          :public => options[:public],
+          :encryption => options[:encryption]
+        }
+      if file.options[:multipart_chunk_size]
```
Comment (henders) [inline lib/remote_files/fog_store.rb]: I should probably check that this relates to the AWS provider in this conditional.

File: lib/remote_files/fog_store.rb [hunk 2]
```diff
@@ -98,5 +95,28 @@ def create_directory
...
+        chunk_size = file.options[:multipart_chunk_size]
+        store_options[:multipart_chunk_size] = chunk_size
+        raise RemoteFiles::Error.new("Minimum chunk size is #{MULITPART_MIN_SIZE}") if chunk_size < MULITPART_MIN_SIZE
+        if !file.content.respond_to?(:read) || !file.content.respond_to?(:size)
+          raise RemoteFiles::Error.new(':content must be a stream if chunking enabled')
```
Comment (grosser) [inline lib/remote_files/fog_store.rb]: does a stream know it's size ?

File: test/fog_store_test.rb [hunk 3]
```diff
@@ -61,6 +61,50 @@
       @store.directory.destroy
       proc { @store.store!(@file) }.must_raise(RemoteFiles::Error)
     end
+
+    it 'should validate multipart uploads meet minimum chunk size' do
+      file = RemoteFiles::File.new('identifier',
```
Comment (grosser) [inline test/fog_store_test.rb]: 2-space indent ? :)

**Comment by grosser** [comment]
:+1: did you try it / does it work ?

File: lib/remote_files/fog_store.rb (hunk 2 above)
Comment (henders) [inline lib/remote_files/fog_store.rb]: Well, File and StringIO have size().  Not sure what you'd normally pass in here besides those, and I wanted a clean way to catch the AWS error with the max 10000 parts. 
Did you want it be an optional check? So if the stream responds to 'size' then do the verification?

**Comment by henders** [comment]
@grosser 
Yup, I tried it, uploaded a 6GB file. Took about an hour!

File: test/fog_store_test.rb (hunk 3 above)
Comment (henders) [inline test/fog_store_test.rb]: Hah, someday this'll become automatic! :)

File: lib/remote_files/fog_store.rb (hunk 2 above)
Comment (grosser) [inline lib/remote_files/fog_store.rb]: meh if it works with files should be fine, was just thinking you might have
an input with unknown size ... but files is good enough :)

**Pull request merged**